pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTableTenantConfigs.java

```diff
@@ -49,12 +50,18 @@
   @ApiResponses(value = {@ApiResponse(code=200, message = "Success"),
       @ApiResponse(code = 404, message = "Table not found"),
       @ApiResponse(code = 500, message = "Internal error rebuilding broker resource or serializing response")})
-  public String rebuildBrokerResource(
+  public SuccessResponse rebuildBrokerResource(
       @ApiParam(value = "Table name (with type)", required = true) @PathParam("tableName") String tableNameWithType
-  ) throws JSONException {
+  ) {
     final PinotResourceManagerResponse pinotResourceManagerResponse =
         _helixResourceManager.rebuildBrokerResourceFromHelixTags(tableNameWithType);
 
-    return pinotResourceManagerResponse.toJSON().toString();
+    if (pinotResourceManagerResponse.isSuccessful()) {
+      return new SuccessResponse(pinotResourceManagerResponse.getMessage());
+    } else if (pinotResourceManagerResponse.getMessage().contains("does not exist")) {
```

Comment:
Either throw the exception from `rebuildBrokerResourceFromHelixTags()` and catch here, or you can throw one kind of exception for all errors. Or, you can move table check here (Many ways to achieve this). Let's not catch it by looking at the message.

Reply:
+1
throw a controllerApplicationException with the right message and status code.